fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/RecordMetaDataBuilder.java

```diff
@@ -435,6 +444,53 @@ public RecordMetaDataBuilder setRecords(@Nonnull Descriptors.FileDescriptor file
         return this;
     }
 
+    /**
+     * Update the records descriptor of the record meta-data.
+     *
+     * <p>
+     * This involves adding new record types and updating descriptors for the existing record types and union fields.
+     * By contract, the extension options will be processed for all of the new record types and will not be processed
+     * for any of the old record types. Also, it is not allowed to call this method when the local file descriptor is set.
+     * </p>
+     *
+     * @param recordsDescriptor the new record descriptor
+     */
+    public void updateRecords(@Nonnull Descriptors.FileDescriptor recordsDescriptor) {
+        updateRecords(recordsDescriptor, true);
+    }
+
+    /**
+     * Update the records descriptor of the record meta-data.
+     *
+     * <p>
+     * This involves adding new record types and updating descriptors for the existing record types and union fields.
```

Comment:
```suggestion
     * This adds any new record types and updates the descriptors for existing record types.
```

I think we want to keep the union fields updating as an internal implementation detail. I could be wrong though.